src/main/kotlin/uk/gov/justice/digital/assessments/restclient/AssessmentApiRestClient.kt

```diff
@@ -93,6 +109,85 @@ class AssessmentApiRestClient {
       .block().also { log.info("Retrieved OASys filtered reference data for $fieldName") }
   }
 
+  fun getOASysRBACPermissions(
+    roleChecks: Set<Roles>,
+    offenderPk: Long? = null,
+    oasysSetPk: Long? = null,
+    assessmentType: AssessmentType? = null,
+    roleNames: Set<RoleNames>? = emptySet()
+  ): RBACPermissionsPayload? {
+    val area = RequestData.getAreaCode()
+    val oasysUserCode = userDetailsRedisRepository.findByUserId(RequestData.getUserId()).oasysUserCode
+
+    log.info("Retrieving OASys RBAC permissions for oasys user $oasysUserCode, area $area, offender $offenderPk, assessment $oasysSetPk and assessment type $assessmentType")
+    val path = "/authorisation/permissions"
+    val permissionsDto = OASysRBACPermissionsDto(
+      oasysUserCode,
+      roleChecks,
+      area,
+      offenderPk,
+      oasysSetPk,
+      assessmentType,
+      roleNames
+    )
+    return webClient
+      .post(
+        path,
+        permissionsDto
+      )
+      .retrieve()
+      .onStatus(HttpStatus::is4xxClientError) {
+        handlePermissionsError(permissionsDto, it, HttpMethod.POST, path)
+      }
+      .onStatus(HttpStatus::is5xxServerError) {
+        handle5xxError(
+          "Failed to retrieve OASys RBAC permissions for oasys user $oasysUserCode, area $area, offender $offenderPk, assessment $oasysSetPk and assessment type $assessmentType",
+          HttpMethod.GET,
+          path,
+          ExternalService.ASSESSMENTS_API
+        )
+      }
+      .bodyToMono(RBACPermissionsPayload::class.java)
+      .block().also { log.info("Retrieved OASys RBAC permissions for oasys user $oasysUserCode, area $area, offender $offenderPk, assessment $oasysSetPk and assessment type $assessmentType") }
+  }
+
+  private fun handlePermissionsError(
+    oASysRBACPermissionsDto: OASysRBACPermissionsDto,
+    clientResponse: ClientResponse,
+    method: HttpMethod,
+    url: String
+  ): Mono<out Throwable?>? {
+    return when (clientResponse.statusCode()) {
+      HttpStatus.FORBIDDEN -> {
+        log.error("Oasys returned Forbidden for $oASysRBACPermissionsDto")
+        clientResponse.bodyToMono(OASysRBACErrorResponse::class.java)
+          .map { error ->
+            ExternalApiForbiddenException(
+              msg = error.developerMessage ?: "",
```

Comment:
is developerMessage non nullable?

Reply:
I thought it could be, and thought that the ?: did that exactly.

`If the expression to the left of ?: is not null, the elvis operator returns it, otherwise it returns the expression to the right. Note that the right-hand side expression is evaluated only if the left-hand side is null.`